### ISSUE TITLE
spectrum_analyzer_x1.ttl from lsp-plugins-lv2-1.1.14-Linux-i586.tar.gz fails to load

### ISSUE THREAD

**Issue by Randrianasulu** [sadko4u/lsp-plugins#85]
cin
Cinelerra Infinity - built: Mar 21 2020 04:59:46
git://git.cinelerra-gg.org/goodguy/cinelerra.git
(c) 2006-2019 Heroine Virtual Ltd. by Adam Williams
2007-2020 mods for Cinelerra-GG by W.P.Morrow aka goodguy
Cinelerra is free software, covered by the GNU General Public License,
and you are welcome to change it and/or distribute copies of it under
certain conditions. There is absolutely no warranty for Cinelerra.

int MWindow::load_filenames(ArrayList<char*>*, int, int, int):
Ошибка открытия /dev/shm/avci_new.mp4
Cinelerra Infinity - built: Mar 21 2020 04:59:46
git://git.cinelerra-gg.org/goodguy/cinelerra.git
(c) 2006-2019 Heroine Virtual Ltd. by Adam Williams
2007-2020 mods for Cinelerra-GG by W.P.Morrow aka goodguy
Cinelerra is free software, covered by the GNU General Public License,
and you are welcome to change it and/or distribute copies of it under
certain conditions. There is absolutely no warranty for Cinelerra.

init plugin index: /usr/lib/cin/plugins
PluginFFilter::new_ffilter(anlms)
  err: Ошибка ввода/вывода
PluginFFilter::new_ffilter(arnndn)
  err: Недопустимый аргумент
PluginFFilter::new_ffilter(axcorrelate)
  err: Ошибка ввода/вывода
[dnn_processing @ 0xe615940] model file for network is not specified
PluginFFilter::new_ffilter(dnn_processing)
  err: Недопустимый аргумент
PluginFFilter::new_ffilter(freezeframes)
  err: Ошибка ввода/вывода
PluginFFilter::new_ffilter(maskedmax)
  err: Ошибка ввода/вывода
PluginFFilter::new_ffilter(maskedmin)
  err: Ошибка ввода/вывода
PluginFFilter::new_ffilter(overlay_opencl)
  err: Ошибка ввода/вывода
PluginFFilter::new_ffilter(xfade)
  err: Ошибка ввода/вывода
PluginFFilter::new_ffilter(xfade_opencl)
  err: Ошибка ввода/вывода
[openclsrc_705 @ 0xe616240] OpenCL source requires output dimensions to be specified.
PluginFFilter::new_ffilter(openclsrc)
  err: Недопустимый аргумент
init lv2 index:
LOAD: http://lsp-plug.in/plugins/lv2/comp_delay_mono
LOAD: http://lsp-plug.in/plugins/lv2/comp_delay_stereo
LOAD: http://lsp-plug.in/plugins/lv2/compressor_lr
LOAD: http://lsp-plug.in/plugins/lv2/compressor_mono
LOAD: http://lsp-plug.in/plugins/lv2/compressor_ms
LOAD: http://lsp-plug.in/plugins/lv2/compressor_stereo
LOAD: http://lsp-plug.in/plugins/lv2/dyna_processor_lr
LOAD: http://lsp-plug.in/plugins/lv2/dyna_processor_mono
LOAD: http://lsp-plug.in/plugins/lv2/dyna_processor_ms
LOAD: http://lsp-plug.in/plugins/lv2/dyna_processor_stereo
LOAD: http://lsp-plug.in/plugins/lv2/expander_lr
LOAD: http://lsp-plug.in/plugins/lv2/expander_mono
LOAD: http://lsp-plug.in/plugins/lv2/expander_ms
LOAD: http://lsp-plug.in/plugins/lv2/expander_stereo
LOAD: http://lsp-plug.in/plugins/lv2/gate_lr
LOAD: http://lsp-plug.in/plugins/lv2/gate_mono
LOAD: http://lsp-plug.in/plugins/lv2/gate_ms
LOAD: http://lsp-plug.in/plugins/lv2/gate_stereo
LOAD: http://lsp-plug.in/plugins/lv2/graph_equalizer_x16_mono
LOAD: http://lsp-plug.in/plugins/lv2/graph_equalizer_x32_mono
LOAD: http://lsp-plug.in/plugins/lv2/impulse_responses_mono
LOAD: http://lsp-plug.in/plugins/lv2/impulse_responses_stereo
LOAD: http://lsp-plug.in/plugins/lv2/impulse_reverb_mono
LOAD: http://lsp-plug.in/plugins/lv2/impulse_reverb_stereo
LOAD: http://lsp-plug.in/plugins/lv2/latency_meter
LOAD: http://lsp-plug.in/plugins/lv2/limiter_mono
LOAD: http://lsp-plug.in/plugins/lv2/limiter_stereo
LOAD: http://lsp-plug.in/plugins/lv2/mb_compressor_mono
LOAD: http://lsp-plug.in/plugins/lv2/mb_expander_lr
LOAD: http://lsp-plug.in/plugins/lv2/mb_expander_mono
LOAD: http://lsp-plug.in/plugins/lv2/mb_expander_ms
LOAD: http://lsp-plug.in/plugins/lv2/mb_expander_stereo
LOAD: http://lsp-plug.in/plugins/lv2/multisampler_x12
LOAD: http://lsp-plug.in/plugins/lv2/multisampler_x12_do
LOAD: http://lsp-plug.in/plugins/lv2/multisampler_x24
LOAD: http://lsp-plug.in/plugins/lv2/multisampler_x24_do
LOAD: http://lsp-plug.in/plugins/lv2/multisampler_x48
LOAD: http://lsp-plug.in/plugins/lv2/multisampler_x48_do
LOAD: http://lsp-plug.in/plugins/lv2/oscillator_mono
LOAD: http://lsp-plug.in/plugins/lv2/phase_detector
LOAD: http://lsp-plug.in/plugins/lv2/profiler_mono
LOAD: http://lsp-plug.in/plugins/lv2/profiler_stereo
LOAD: http://lsp-plug.in/plugins/lv2/room_builder_mono
LOAD: http://lsp-plug.in/plugins/lv2/room_builder_stereo
LOAD: http://lsp-plug.in/plugins/lv2/sampler_mono
LOAD: http://lsp-plug.in/plugins/lv2/sampler_stereo
LOAD: http://lsp-plug.in/plugins/lv2/sc_compressor_mono
LOAD: http://lsp-plug.in/plugins/lv2/sc_compressor_ms
LOAD: http://lsp-plug.in/plugins/lv2/sc_compressor_stereo
LOAD: http://lsp-plug.in/plugins/lv2/sc_expander_lr
LOAD: http://lsp-plug.in/plugins/lv2/sc_expander_mono
LOAD: http://lsp-plug.in/plugins/lv2/sc_expander_ms
LOAD: http://lsp-plug.in/plugins/lv2/sc_expander_stereo
LOAD: http://lsp-plug.in/plugins/lv2/sc_gate_lr
LOAD: http://lsp-plug.in/plugins/lv2/sc_gate_mono
LOAD: http://lsp-plug.in/plugins/lv2/sc_gate_ms
LOAD: http://lsp-plug.in/plugins/lv2/sc_gate_stereo
LOAD: http://lsp-plug.in/plugins/lv2/sc_limiter_mono
LOAD: http://lsp-plug.in/plugins/lv2/sc_limiter_stereo
LOAD: http://lsp-plug.in/plugins/lv2/sc_mb_expander_lr
LOAD: http://lsp-plug.in/plugins/lv2/sc_mb_expander_mono
LOAD: http://lsp-plug.in/plugins/lv2/sc_mb_expander_ms
LOAD: http://lsp-plug.in/plugins/lv2/sc_mb_expander_stereo
LOAD: http://lsp-plug.in/plugins/lv2/slap_delay_mono
LOAD: http://lsp-plug.in/plugins/lv2/slap_delay_stereo
LOAD: http://lsp-plug.in/plugins/lv2/spectrum_analyzer_x1
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: missing ';' or '.'
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: expected `]', not `l'
lilv_world_load_file(): error: Error loading file `file:///usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl'
lv2: lilv_plugin_instantiate failed
LOAD: http://lsp-plug.in/plugins/lv2/spectrum_analyzer_x12
LOAD: http://lsp-plug.in/plugins/lv2/spectrum_analyzer_x16
LOAD: http://lsp-plug.in/plugins/lv2/spectrum_analyzer_x2
LOAD: http://lsp-plug.in/plugins/lv2/spectrum_analyzer_x4
LOAD: http://lsp-plug.in/plugins/lv2/spectrum_analyzer_x8
LOAD: http://lsp-plug.in/plugins/lv2/trigger_mono
LOAD: http://lsp-plug.in/plugins/lv2/trigger_stereo
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: missing ';' or '.'
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: expected `]', not `l'
lilv_world_load_file(): error: Error loading file `file:///usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl'
lv2: lilv_plugin_instantiate failed
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: missing ';' or '.'
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: expected `]', not `l'
lilv_world_load_file(): error: Error loading file `file:///usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl'
lv2: lilv_plugin_instantiate failed
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: missing ';' or '.'
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: expected `]', not `l'
lilv_world_load_file(): error: Error loading file `file:///usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl'
lv2: lilv_plugin_instantiate failed
lv2ui: unable to init: http://lsp-plug.in/plugins/lv2/spectrum_analyzer_x1
ALSA lib confmisc.c:1281:(snd_func_refer) Unable to find definition 'defaults.pcm.dmix.channels'
ALSA lib conf.c:4568:(_snd_config_evaluate) function snd_func_refer returned error: Нет такого файла или каталога
ALSA lib conf.c:5036:(snd_config_expand) Args evaluate error: Нет такого файла или каталога
ALSA lib pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM dmix:SB
AudioALSA::open_output default: Нет такого файла или каталога
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: missing ';' or '.'
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: expected `]', not `l'
lilv_world_load_file(): error: Error loading file `file:///usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl'
lv2: lilv_plugin_instantiate failed
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: missing ';' or '.'
error: /usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: expected `]', not `l'
lilv_world_load_file(): error: Error loading file `file:///usr/share/cin/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl'
lv2: lilv_plugin_instantiate failed
lv2ui: unable to init: http://lsp-plug.in/plugins/lv2/spectrum_analyzer_x1
Session time: 0:02:18
Cpu time: user: 0:00:31.329 sys: 0:00:05.683

Host is Cinelerra-GG compiled from git. I have some other issues with way it compiles LV2-related libs, it seems, but I'll try to figure this piece of puzzle separately

**Comment by sadko4u** [comment]
That's sad error. Can be already fixed by this merge-commit: fe9b21a2d378eccbe26c7f0835fa614f6c0f6cc8

**Comment by sadko4u** [comment]
Cherry-picked changes to **hotfix** branch

**Comment by Vascom** [comment]
Seems I have same bug with pulseeffects
```
error: /usr/lib64/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: missing ';' or '.'
error: /usr/lib64/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl:278:2: expected `]', not `l'
lilv_world_load_file(): error: Error loading file `file:///usr/lib64/lv2/lsp-plugins.lv2/spectrum_analyzer_x1.ttl'
```

**Comment by sadko4u** [comment]
@Vascom 
Yes, that's sad issue: I've a bit broken TTL file generator when implemented localization features.
You may manually patch TTL file and add missing semicolon or build LSP Plugins from hotfix branch.

**Comment by Vascom** [comment]
Is 1.1.15 with fixes planned soon?

**Comment by sadko4u** [comment]
I believe I'll issue the fix within week.
There may another problems appear for other users (but I hope they won't), so let's have a small time gap between 1.1.14 and 1.1.15 release.

**Comment by Vascom** [comment]
OK.

**Comment by sadko4u** [comment]
For users plugins didn't change a lot but the code base changes significantly from one release to other. And it's very hard to cover all cases which may break. I'm trying to cover new modules and functions with unit tests but this slows down the development **significantly**.

**Comment by sadko4u** [comment]
Available in 1.1.15 release!
